Sunpy v1

ianan opened the pull request:
Version that works with SunPy v1 and v2 and adds in extra options for filtering (by time, position and det_id).

ianan committed:
Merge pull request #2 from NuSTAR/master

update

ianan committed:
Merge pull request #3 from NuSTAR/master

sync from original

ianan committed:
Merge pull request #4 from NuSTAR/master

update my fork

ianan committed:
map fixed for v1

ianan committed:
map tweaks with v1

ianan committed:
filter with time and xy

ianan committed:
planning utils convert now for v1

ianan committed:
fixed histogram2d bug and tested with v2

ianan committed:
filter with det_id

ayshih commented on nustar_pysolar/map.py:
```diff
@@ -94,18 +104,31 @@ def make_sunpy(evtdata, hdr, norm_map=False):
     # Assumes dsun_obs in m if don't specify the units, so give units
-    "DSUN_OBS": get_sunearth_distance(mid_obs_time).value*u.astrophys.au
+#    SunPy v1+ version        
+    "DSUN_OBS": sunpy.coordinates.sun.earth_distance(mid_obs_time).value*u.astrophys.au
```
```suggestion
    "DSUN_OBS": sunpy.coordinates.sun.earth_distance(mid_obs_time).to('AU')
```
Alternatively, if having `DSUN_OBS` as a number with units causes any issues, you can instead just get its value in meters: `sunpy.coordinates.sun.earth_distance(mid_obs_time).to_value('m')`.

ayshih commented on nustar_pysolar/map.py:
```diff
@@ -94,18 +104,31 @@ def make_sunpy(evtdata, hdr, norm_map=False):
+#    SunPy <v1 version
+#    "RSUN_OBS": sun.solar_semidiameter_angular_size(mid_obs_time).value,
+##        
+#    sun.constants will stay in SunPy v1+ ? just sun.sun being depreciated?
```
Yes, `sunpy.sun.contants` is sticking around for the foreseeable future.

ayshih commented:
For the record, `sun.P()` returns the position angle of the Sun specifically relative to true Earth north, not to celestial north.  This code appears to use it as the latter.  Due to polar motion of the Earth's axis, the discrepancy can be up to ~10 arcseconds in the position angle, or up to ~0.05 arcsec in translation at the solar limb.  If that level of error is acceptable, then carry on.  (If not, I can help you make the code more accurate.)

ianan commented:
That might be useful but if it is much less than an arcsecond in x,y that won't do much to the NuSTAR maps.

bwgref commented:
Re: sun.P() I don't think this matter for the work here since we're using ~2.5 arc second sky pixels and I think the miss for a 10-arcsec rotation at about R_sun is a fraction of an arc second. We use this to determine (a) the rotation of the NuSTAR FoV planning purposes (which only needs to be accurate to a degree anyways) and (b) the rotation matrix to convert the NuSTAR X/Y sky coordinates (with are effectively J2000 RA/Dec) to be wrt solar north. I guess there's also the question of whether there's another step when going from J2000 north to "current epoch" north as well. I think it's still going to be a small effect for us, but maybe I should go remember how much the RA/Dec coordinate system moves in 20 years...

Also, I'll try to see what happened with the CI failure. Looks like a numpy versioning thing in the unit tests.

bwgref committed:
Package maintanence

Updated conftest.py to match current standards.
Streamlined requirements.txt
Updated setup.py with 0.4dev version number.

bwgref commented:
Okay, CI failures look like a TravisCI lag or something. My local pytest passes just, so I'm going to ahead and merge.

bwgref committed:
Update nustar_pysolar/map.py

Co-authored-by: Albert Y. Shih <ayshih@gmail.com>
